File: 6_kyu/Highest rank number in an array/solution.rb
def highest_rank(arr)
  freqs = {}
  arr.each do |num|
    freqs[num] ? freqs[num] += 1 : freqs[num] = 1
  end
  sorted = freqs.sort_by(&:reverse).to_h
  sorted.keys.last.to_i
end

File: 6_kyu/Highest rank number in an array/solution.js
function highestRank(arr){
  const freqs = {};
  for (const num of arr) {
    freqs[num] ? freqs[num] += 1 : freqs[num] = 1;
  }
  const sorted = Object.entries(freqs).sort((a, b) => a[1] - b[1]);
  return Number(sorted.pop()[0]);
}
